collector/generatorreceiver/internal/generator/trace_generator.go

```diff
@@ -54,11 +54,10 @@ func (g *TraceGenerator) genSpanId() pdata.SpanID {
 	return pdata.NewSpanID(traceId)
 }
 
-func (g *TraceGenerator) Generate(startTimeMicros int64) *pdata.Traces {
-	rootService := g.topology.GetServiceTier(g.service)
+func (g *TraceGenerator) Generate(startTimeNanos int64) *pdata.Traces {
 	traces := pdata.NewTraces()
 
-	g.createSpanForServiceRouteCall(&traces, rootService, g.route, startTimeMicros, g.genTraceId(), pdata.NewSpanID([8]byte{0x0}))
+	g.createSpanForServiceRouteCall(&traces, g.service, g.route, startTimeNanos, g.genTraceId(), pdata.NewSpanID([8]byte{0x0}))
```

Comment:
Nice, I like the consistency of having most of our methods use name strings rather than the object (pointer) where possible.